sdk/core/core-amqp/src/errors.ts

```diff
@@ -464,9 +468,23 @@ export class MessagingError extends Error {
   info?: any;
   /**
    * @param {string} message The error message that provides more information about the error.
+   * @param originalError An error whose properties will be copied to the MessagingError if the
+   * propert doesn't already exist.
    */
-  constructor(message: string) {
+  constructor(message: string, originalError?: Error) {
     super(message);
+
+    // copy properties from the original error.
+    if (!originalError) {
+      return;
+    }
+
+    // can consider using a mixin instead but not sure this gives us much value over casting to any
```

Comment:
Yeah... not so much a big fan of the sloppy types here. It seems good to know which properties we will copy and which we won't. Not sure what you mean by mixin though?

Reply:
I struggled with what to do here since I wanted to make sure we preserved anything the original error may have had even if we didn't know about it.

That said, we're now only converting AmqpErrors and [SystemError](https://nodejs.org/dist/latest-v12.x/docs/api/errors.html#errors_class_systemerror), so I could add the properties from SystemError to MessagingError as optional properties:
> - address <string> If present, the address to which a network connection failed
> - code <string> The string error code
> - dest <string> If present, the file path destination when reporting a file system error
> - errno <number> | <string> The system-provided error number
> - info <Object> If present, extra details about the error condition
> - message <string> A system-provided human-readable description of the error
> - path <string> If present, the file path when reporting a file system error
> - port <number> If present, the network connection port that is not available
> - syscall <string> The name of the system call that triggered the error

Reply:
I got rid of the use of `any` here.